src/main/java/com/xeeva/navigation/MainNav.java

```diff
@@ -1,53 +1,313 @@
+	 * Quantity,Unit Price
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void verify_UpdateCart(String UP, String UOM, String Qty,
+			int cartItemRows) {
+
+		// Clicking on Edit
+		driver.findElement(	By.xpath(".//*[@id='tblCartInfo']/tbody/tr["+ (cartItemRows - 1) + "]/td/div/a[2]")).click();
+		Sleeper.sleep(3000);
+
+		// Providing Update Values to Smart Form
+		TestReporter.logStep(itemDescription.getAttribute("value"));
+		txtQuantity.safeSet(Qty);
+		unitOfMeasure.select(UOM);
+		txtUnitPrice.safeSet(UP);
+		btnSubmit.click();
+
+		// Clicking on Cart Link
+		clickCartItemsLink();
+
+		// Clicking on Edit
+		driver.findElement(
+				By.xpath(".//*[@id='tblCartInfo']/tbody/tr["
+						+ (cartItemRows - 1) + "]/td/div/a[2]")).click();
+		Sleeper.sleep(3000);
+
+		// Verifying Cart Update
+		TestReporter.assertTrue(txtQuantity.getAttribute("value").equalsIgnoreCase(Qty), "Smart Item Qunatity Updated!!");
+		TestReporter.assertTrue(txtUnitPrice.getAttribute("value").equalsIgnoreCase(UP), "Unit Price Updated!!");
+		TestReporter.assertTrue(unitOfMeasure.getFirstSelectedOption().getText().equals(UOM), "Unit Of measure Updated!!");
+
+	}
+
+	/**@summary This Method performs Verifications For Smart Form Items 
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void verifyCartItem(String ItemDescription, String UnitPrice,
+			String UOM, String Quantity) {
+
+		// clicking cart link
+		clickCartItemsLink();
+
+		// Reading Rows/Columns Count from Cart Item grid
+		List<WebElement> cartItemRows = driver.findWebElements(By.xpath(".//*[@id='tblCartInfo']/tbody/tr"));
+		TestReporter.logStep("Total Rows available in Cart :"+ cartItemRows.size());
+		List<WebElement> cartItemCols = driver.findElements(By.xpath(".//*[@id='tblCartInfo']/tbody/tr["
+				+ (cartItemRows.size() - 1) + "]/td"));
+		TestReporter.logStep("Total Columns available in Cart :"+ cartItemCols.size());
+
+		// Verification - Smart form Item Updated with Quantity, UOM and UnitPrice
+		verify_UpdateCart(UnitPrice, UOM, Quantity, cartItemRows.size());
+
+	}
+
+	/**
+	 * @summary This Method performs all Verifications For  Local,Global,Favorite,ComparisonSceen Cart Items
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void perform_CartItemVerifications(String UnitPrice,String UOM,String Quantity,String itemNumber) {
+		// clicking cart link
+		clickCartItemsLink();
+
+		// Reading Rows Count from CartItem grid
+		List<WebElement> cartItemRows = driver.findWebElements(By
+				.xpath(".//*[@id='tblCartInfo']/tbody/tr"));
+		TestReporter.logStep("Total Rows available in Cart :"+ cartItemRows.size());
+
+		// Click on Edit
+		click_CartItemEdit(itemNumber);
+
+		// Update - Unit Price , UOM
+		TestReporter.logStep("Item Qty befor Update : "+ txtItemQuantity.getAttribute("value"));
+		String actualQty = txtItemQuantity.getAttribute("value");
+
+		Sleeper.sleep(3000);
+		txtItemUnitPrice.isDisplayed();

```

Comment:
line 218 - the isDisplayed() method doesn't wait for something to be displayed, only returns a boolean of if the element is displayed.  You might want to do a .syncvisible instead.